internal/procedures/server/start.go

```diff
@@ -74,3 +78,57 @@ func startCourse(course *model.Course) {
 		log.Error("Failed to update course.", err, course)
 	}
 }
+
+// Check to see if a server is running and start one if it's not.
+// Returns (false, 0) if a the primary server or cmd test server is already running
+// or (true, oldPort) if it started it's own server.
+func MustEnsureServerIsRunning() (bool, int) {
```

Comment:
!!

I don't like this here, and I am concerned that you moved it here.
The previous comment on this function was that it should be in another file.
I didn't say another **package**.

This may seem like a small thing, but I see this as an orange flag (very close to red).
This function is not used by anything in this package and now has to be exported.
It's already a "must" function, so we would want to only export if absolutely necessary (which it is as it currently stands since it is in the wrong package).